### PR TITLE
Document JULIA_CPU_TARGET in environment-variables.md

### PR DESCRIPTION
**Pull request by mkitti**
Document `JULIA_CPU_TARGET` in the Environment Variables section.

https://docs.julialang.org/en/v1/manual/environment-variables/#Environment-Variables

### CODE REVIEW

**Review by simonbyrne — APPROVED** [review]

**Comment by jishnub** [comment]
Perhaps worth mentioning that this is of particular importance on heterogeneous systems?

File: doc/src/manual/environment-variables.md [hunk 1]
```diff
@@ -376,6 +376,12 @@ should have at the terminal.
 The formatting `Base.answer_color()` (default: normal, `"\033[0m"`) that output
 should have at the terminal.
 
+## System and Package Image Building
+
+### `JULIA_CPU_TARGET`
+
+Modify the target machine architecture for (pre)compiling system and package images. See [`System image optimized for multiple microarchitectures`](@ref sysimg-multi-versioning)
```
Comment (vchuravy) [inline doc/src/manual/environment-variables.md:383]: You might also want to link to the pkgimage docs, but crucially these are currently in the dev-docs and we don;t have a good "user-level" section that talks about precompilation & caching.

cc: @timholy

**Comment by mkitti** [comment]
I borrowed some additional text from https://github.com/mkitti/julia/blob/0a05a5b05d382bdf38da7fdd0c069dff8bdc072c/src/processor.cpp#L26-L80

I added an explict note for HPC systems and linked package images.

File: doc/src/manual/environment-variables.md [hunk 2]
```diff
@@ -376,6 +376,53 @@ should have at the terminal.
 The formatting `Base.answer_color()` (default: normal, `"\033[0m"`) that output
 should have at the terminal.
 
+## System and Package Image Building
```
Comment (vchuravy) [inline doc/src/manual/environment-variables.md:379]: This is great! Thank you. My only comment is that it is odd to place it in the environment-variables section.

Maybe a better place would be under https://docs.julialang.org/en/v1.10-dev/manual/code-loading/?

Comment (mkitti) [reply doc/src/manual/environment-variables.md:379]: We do need a reference in the Environment Variables section for `JULIA_CPU_TARGET` though, so I guess the question is where should the details go. I initially started with a few minimal notes here and then expanded.

Here I've tried to focus primarily on what does the environment variable does and what are the valid settings for it.

In what I copied, I specifically omitted this part of the comments that deal with loading and dispatch:
https://github.com/mkitti/julia/blob/0a05a5b05d382bdf38da7fdd0c069dff8bdc072c/src/processor.cpp#L31-L54
I think a dicussion of that part should go into code loading... or perhaps a new section called "code generation".

In this pull request, I would like to focus on documenting the environment variable. If there are too many details here, I would be happy to edit it down.

Comment (mkitti) [reply doc/src/manual/environment-variables.md:379]: @simonbyrne , where you would expect to find this documentation?

Comment (ViralBShah) [reply doc/src/manual/environment-variables.md:379]: Should it be in code loading, with a link in environment variables - or the other way around? I do feel it is good to have either fully or as a link in the env variables section.

Comment (mkitti) [reply doc/src/manual/environment-variables.md:379]: If it pertains the value of `JULIA_CPU_TARGET` it should be documented here.

A detailed discussion of code loading can be expanded in a separate section.

I think this pull request should focus on the environment variable section. I think another pull request by someone with a better understanding of code loading would be appropriate, but I would be willing to take a shot at it.

**Comment by ViralBShah** [comment]
@vchuravy Ok to merge?

**Review by vchuravy — APPROVED** [review]

**Pull request merged**